File: src/main/java/net/hectus/neobb/turn/default_game/mob/TPolarBear.java
package net.hectus.neobb.turn.default_game.mob;

import net.hectus.neobb.buff.Buff;
import net.hectus.neobb.player.NeoPlayer;
import net.hectus.neobb.turn.default_game.attributes.clazz.ColdClazz;
import net.hectus.neobb.turn.default_game.attributes.function.BuffFunction;
import org.bukkit.entity.PolarBear;
import org.bukkit.potion.PotionEffectType;

import java.util.List;

public class TPolarBear extends MobTurn<PolarBear> implements BuffFunction, ColdClazz {
    public TPolarBear(NeoPlayer player) { super(player); }
    public TPolarBear(PolarBear data, NeoPlayer player) { super(data, player); }

    @Override
    public int cost() {
        return 4;
    }

    @Override
    public void apply() {
        player.addModifier("no_jump");
    }

    @Override
    public List<Buff> buffs() {
        return List.of(new Buff.Effect(Buff.BuffTarget.ALL, PotionEffectType.SPEED, 3));
    }
}

File: src/main/kotlin/net/hectus/neobb/modes/turn/default_game/mob/TPolarBear.kt
package net.hectus.neobb.modes.turn.default_game.mob

import com.marcpg.libpg.storing.Cord
import net.hectus.neobb.buff.Buff
import net.hectus.neobb.buff.Effect
import net.hectus.neobb.modes.turn.default_game.attribute.clazz.ColdClazz
import net.hectus.neobb.modes.turn.default_game.attribute.function.BuffFunction
import net.hectus.neobb.player.NeoPlayer
import net.hectus.neobb.util.Modifiers
import org.bukkit.entity.PolarBear
import org.bukkit.potion.PotionEffectType

class TPolarBear(data: PolarBear?, cord: Cord?, player: NeoPlayer?) : MobTurn<PolarBear>(data, cord, player), BuffFunction, ColdClazz {
    override val cost: Int = 4

    override fun apply() {
        player!!.addModifier(Modifiers.Player.NO_JUMP)
    }

    override fun buffs(): List<Buff<*>> {
        return listOf(Effect(PotionEffectType.SPEED, 3, Buff.BuffTarget.ALL))
    }
}
